AssetRipper.Library/Configuration/LibraryConfiguration.cs

```diff
@@ -1,73 +1,40 @@
 		{
 			base.ResetToDefaultValues();
-			AudioExportFormat = AudioExportFormat.Default;
-			ImageExportFormat = ImageExportFormat.Png;
-			MeshExportFormat = MeshExportFormat.Native;
-			ScriptExportMode = ScriptExportMode.Decompiled;
-			ScriptLanguageVersion = ScriptLanguageVersion.AutoSafe;
-			ShaderExportMode = ShaderExportMode.Dummy;
-			SpriteExportMode = SpriteExportMode.Yaml;
-			TerrainExportMode = TerrainExportMode.Yaml;
-			TextExportMode = TextExportMode.Parse;
+
+			this.SetSetting<AudioExportFormat>(AudioExportFormat.Default);
+			this.SetSetting<ImageExportFormat>(ImageExportFormat.Png);
+			this.SetSetting<MeshExportFormat>(MeshExportFormat.Native);
+			this.SetSetting<ScriptExportMode>(ScriptExportMode.Decompiled);
+			this.SetSetting<ScriptLanguageVersion>(ScriptLanguageVersion.AutoSafe);
+			this.SetSetting<ShaderExportMode>(ShaderExportMode.Dummy);
+			this.SetSetting<SpriteExportMode>(SpriteExportMode.Yaml);
+			this.SetSetting<TerrainExportMode>(TerrainExportMode.Yaml);
+			this.SetSetting<TextExportMode>(TextExportMode.Parse);
 		}
 
 		public override void LogConfigurationValues()
 		{
 			base.LogConfigurationValues();
-			Logger.Info(LogCategory.General, $"{nameof(AudioExportFormat)}: {AudioExportFormat}");
-			Logger.Info(LogCategory.General, $"{nameof(ImageExportFormat)}: {ImageExportFormat}");
-			Logger.Info(LogCategory.General, $"{nameof(MeshExportFormat)}: {MeshExportFormat}");
-			Logger.Info(LogCategory.General, $"{nameof(ScriptExportMode)}: {ScriptExportMode}");
-			Logger.Info(LogCategory.General, $"{nameof(ScriptLanguageVersion)}: {ScriptLanguageVersion}");
-			Logger.Info(LogCategory.General, $"{nameof(ShaderExportMode)}: {ShaderExportMode}");
-			Logger.Info(LogCategory.General, $"{nameof(SpriteExportMode)}: {SpriteExportMode}");
-			Logger.Info(LogCategory.General, $"{nameof(TerrainExportMode)}: {TerrainExportMode}");
-			Logger.Info(LogCategory.General, $"{nameof(TextExportMode)}: {TextExportMode}");
+
+			Logger.Info(LogCategory.General, $"{nameof(AudioExportFormat)}: {this.GetSetting<AudioExportFormat>()}");
+			Logger.Info(LogCategory.General, $"{nameof(ImageExportFormat)}: {this.GetSetting<ImageExportFormat>()}");
+			Logger.Info(LogCategory.General, $"{nameof(MeshExportFormat)}: {this.GetSetting<MeshExportFormat>()}");
+			Logger.Info(LogCategory.General, $"{nameof(ScriptExportMode)}: {this.GetSetting<ScriptExportMode>()}");
+			Logger.Info(LogCategory.General, $"{nameof(ScriptLanguageVersion)}: {this.GetSetting<ScriptLanguageVersion>()}");
+			Logger.Info(LogCategory.General, $"{nameof(ShaderExportMode)}: {this.GetSetting<ShaderExportMode>()}");
+			Logger.Info(LogCategory.General, $"{nameof(SpriteExportMode)}: {this.GetSetting<SpriteExportMode>()}");
+			Logger.Info(LogCategory.General, $"{nameof(TerrainExportMode)}: {this.GetSetting<TerrainExportMode>()}");
+			Logger.Info(LogCategory.General, $"{nameof(TextExportMode)}: {this.GetSetting<TextExportMode>()}");
 		}
 	}
```

Comment:
`ResetToDefaultValues` and `LogConfigurationValues` are exactly the kind of methods which should be source generated.

Reply:
Good idea, although since they were like this before, I don't think that's in scope for this pr, since it's about just adding missing options to the cli.